purchases/src/main/kotlin/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -136,7 +138,36 @@ internal class HTTPClient(
         }
         if (callResult == null) {
             log(LogIntent.WARNING, NetworkStrings.ETAG_RETRYING_CALL)
-            callResult = performRequest(baseURL, endpoint, body, postFieldsToSign, requestHeaders, refreshETag = true)
+            callResult = performRequest(
+                baseURL,
+                endpoint,
+                body,
+                postFieldsToSign,
+                requestHeaders,
+                refreshETag = true,
+                fallbackBaseURLs,
+                fallbackURLIndex,
+            )
+        } else if (RCHTTPStatusCodes.isServerError(callResult.responseCode) &&
+            endpoint.supportsFallbackBaseURLs &&
+            fallbackURLIndex in fallbackBaseURLs.indices
```

Comment:
Perf nit, but a comparison is probably faster than a "contains": 
```suggestion
            fallbackURLIndex <= fallbackBaseURLs.lastIndex
```